effekt/shared/src/main/scala/effekt/Typer.scala

```diff
@@ -1454,15 +1467,15 @@ trait TyperOps extends ContextOps { self: Context =>
 
   private [typer] def bindingCapabilities[R](binder: source.Tree, caps: List[symbols.BlockParam])(f: => R): R = {
     bindCapabilities(binder, caps)
-    capabilityScope = BindSome(binder, caps.map { c => c.tpe.asInterfaceType -> c }.toMap, capabilityScope)
+    capabilityScope = BindSome(binder, caps.map { c => c.tpe.get.asInterfaceType -> c }.toMap, capabilityScope)
     val result = f
     capabilityScope = capabilityScope.parent
     result
   }
 
   private [typer] def bindCapabilities[R](binder: source.Tree, caps: List[symbols.BlockParam]): Unit =
     val capabilities = caps map { cap =>
-      assertConcrete(cap.tpe.asInterfaceType)
+      assertConcrete(cap.tpe.get.asInterfaceType)
```

Comment:
Sorry for all those comments, but how do we know that the type exists here?

Reply:
This function is only ever called by `bindingCapabilities` where we already ensured that a type is given. However, I'll also add `.getOrElse { Context.abort ... }` here, just in case later modifications change these guarantees.